events/janus_gelfevh.c

```diff
@@ -0,0 +1,589 @@
+
+#include <math.h>
+
+#include "../debug.h"
+#include "../config.h"
+#include "../mutex.h"
+#include "../utils.h"
+#include "../events.h"
+#include <netdb.h>
+#include <errno.h>
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include "../ip-utils.h"
+
+/* Plugin information */
+#define JANUS_GELFEVH_VERSION			1
+#define JANUS_GELFEVH_VERSION_STRING 	"0.0.1"
+#define JANUS_GELFEVH_DESCRIPTION 		"This is event handler plugin for Janus, which forwards events via TCP/UDP to GELF server."
+#define JANUS_GELFEVH_NAME 				"JANUS GelfEventHandler plugin"
+#define JANUS_GELFEVH_AUTHOR 			"Mirko Brankovic <mirkobrankovic@gmail.com>"
+#define JANUS_GELFEVH_PACKAGE			"janus.eventhandler.gelfevh"
+
+#define MAX_GELF_CHUNKS 				128
+
+/* Plugin methods */
+janus_eventhandler *create(void);
+int janus_gelfevh_init(const char *config_path);
+void janus_gelfevh_destroy(void);
+int janus_gelfevh_get_api_compatibility(void);
+int janus_gelfevh_get_version(void);
+const char *janus_gelfevh_get_version_string(void);
+const char *janus_gelfevh_get_description(void);
+const char *janus_gelfevh_get_name(void);
+const char *janus_gelfevh_get_author(void);
+const char *janus_gelfevh_get_package(void);
+void janus_gelfevh_incoming_event(json_t *event);
+json_t *janus_gelfevh_handle_request(json_t *request);
+
+/* Event handler setup */
+static janus_eventhandler janus_gelfevh =
+	JANUS_EVENTHANDLER_INIT (
+		.init = janus_gelfevh_init,
+		.destroy = janus_gelfevh_destroy,
+
+		.get_api_compatibility = janus_gelfevh_get_api_compatibility,
+		.get_version = janus_gelfevh_get_version,
+		.get_version_string = janus_gelfevh_get_version_string,
+		.get_description = janus_gelfevh_get_description,
+		.get_name = janus_gelfevh_get_name,
+		.get_author = janus_gelfevh_get_author,
+		.get_package = janus_gelfevh_get_package,
+
+		.incoming_event = janus_gelfevh_incoming_event,
+		.handle_request = janus_gelfevh_handle_request,
+
+		.events_mask = JANUS_EVENT_TYPE_NONE
+	);
+
+/* Plugin creator */
+janus_eventhandler *create(void) {
+	JANUS_LOG(LOG_VERB, "%s created!\n", JANUS_GELFEVH_NAME);
+	return &janus_gelfevh;
+}
+
+/* Compression, if any */
+static gboolean compress = FALSE;
+static int compression = 6; /* Z_DEFAULT_COMPRESSION */
+
+/* Useful stuff */
+static volatile gint initialized = 0, stopping = 0;
+static GThread *handler_thread;
+static void *janus_gelfevh_handler(void *data);
+static janus_mutex evh_mutex;
+
+/* JSON serialization options */
+static size_t json_format = JSON_INDENT(3) | JSON_PRESERVE_ORDER;
+
+/* Queue of events to handle */
+static GAsyncQueue *events = NULL;
+static json_t exit_event;
+static void janus_gelfevh_event_free(json_t *event) {
+	if(!event || event == &exit_event)
+		return;
+	json_decref(event);
+}
+
+/* GELF backend to send the events to */
+static char *backend = NULL;
+static char *port = NULL;
+
+typedef enum janus_gelfevh_socket_type {
+	TCP = 1,
+	UDP = 2
+} janus_gelfevh_socket_type;
```

Comment:
Per code style, the enum values should be prefixed too, e.g., `JANUS_GELFEVH_SOCKET_TYPE_UDP`, just to ensure there are not conflicts with anything else.